packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -481,34 +493,42 @@ export class TokenDetectionController extends StaticIntervalPollingController<
       return;
     }
 
-    const addressAgainstWhichToDetect = accountAddress ?? this.#selectedAddress;
-    const {
-      chainId: chainIdAgainstWhichToDetect,
-      networkClientId: networkClientIdAgainstWhichToDetect,
-    } = this.#getCorrectChainIdAndNetworkClientId(networkClientId);
+    this.#addressAgainstWhichToDetect = accountAddress ?? this.#selectedAddress;
+    const { chainId, networkClientId: selectedNetworkClientId } =
+      this.#getCorrectChainIdAndNetworkClientId(networkClientId);
+    this.#chainIdAgainstWhichToDetect = chainId;
+    this.#networkClientIdAgainstWhichToDetect = selectedNetworkClientId;
 
-    if (!isTokenDetectionSupportedForNetwork(chainIdAgainstWhichToDetect)) {
+    if (
+      !isTokenDetectionSupportedForNetwork(this.#chainIdAgainstWhichToDetect)
+    ) {
       return;
     }
-
     if (
       !this.#isDetectionEnabledFromPreferences &&
-      chainIdAgainstWhichToDetect !== ChainId.mainnet
+      this.#chainIdAgainstWhichToDetect !== ChainId.mainnet
     ) {
       return;
     }
     const isTokenDetectionInactiveInMainnet =
       !this.#isDetectionEnabledFromPreferences &&
-      chainIdAgainstWhichToDetect === ChainId.mainnet;
+      this.#chainIdAgainstWhichToDetect === ChainId.mainnet;
     const { tokensChainsCache } = this.messagingSystem.call(
       'TokenListController:getState',
     );
-    const tokenList =
-      tokensChainsCache[chainIdAgainstWhichToDetect]?.data ?? {};
-    const tokenListUsed = isTokenDetectionInactiveInMainnet
+    this.#tokenList = isTokenDetectionInactiveInMainnet
       ? STATIC_MAINNET_TOKEN_LIST
-      : tokenList;
+      : tokensChainsCache[this.#chainIdAgainstWhichToDetect]?.data ?? {};
```

Comment:
[Consolidate tokenList, tokenListUsed variables](https://github.com/MetaMask/core/pull/3938/commits/9b6d9cea3d48538dcff8b5ccfc21e4eaf0b1b20f), and assign to class field so it can be used in the refactored `detectTokens` helper methods.